qiskit/_result.py

```diff
@@ -142,14 +140,17 @@ def get_data(self, name):
 
         Returns:
             A dictionary of data for the different backends.
+            If there's an error, the function will throw.
         """
+        if self._is_error():
+            raise self.__result['result'] # Rethrow exception!
+
         try:
             qobj = self.__qobj
             for index in range(len(qobj['circuits'])):
                 if qobj['circuits'][index]['name'] == name:
                     return self.__result['result'][index]['data']
-        except (KeyError, TypeError) as err:
-            print(err)
+        except (KeyError, TypeError):
             raise QISKitError('No data for circuit "{0}"'.format(name))
```

Comment:
Looking through the code and maybe not really related to this PR - would it make sense to rename or do something about `_results.ResultError`? It seems a bit counterintuitive that it is declared inside this module, but seems to used only for raising errors on remote backends and not for exceptions inside a `Result`.

Reply:
I'm gonna move ResultError out of _result.py to it's own file: _resulterror.py. ResultError is basically a wrapper over backend error results: { status = 'error', message='....', code='ONE_OF_THE_CODES' }. I don't think exposing it to the user will add some value, so it will be used internally. The user will receive a QISKitError eventually, with an error message explaining what happened.